h_malloc.c

```diff
@@ -1368,6 +1477,11 @@ EXPORT void *h_calloc(size_t nmemb, size_t size) {
     if (!ZERO_ON_FREE && likely(p != NULL) && total_size && total_size <= max_slab_size_class) {
         memset(p, 0, total_size - canary_size);
     }
+#ifdef HAS_ARM_MTE
+    // use an assert instead of adding a conditional to memset() above (freed memory is always
+    // zeroed when MTE is enabled)
```

Comment:
```suggestion
    // use an assert instead of adding a conditional to memset() above (freed memory is always
    // zeroed, even when MTE is enabled)
```

Reply:
Not sure what you mean. Freed memory is not zeroed when ZERO_ON_FREE is false and MTE is disabled (either statically or at runtime).
I've added this assert to skip adding an extra runtime conditional to memset() above.

Reply:
```suggestion
    // use an assert instead of adding a conditional to memset() above, since freed memory is always
    // zeroed when MTE is enabled 
```